Sources/MongoSwift/Operations/RenameCollectionOperation.swift

```diff
@@ -0,0 +1,46 @@
+import CLibMongoC
+
+/// An operation corresponding to a "createIndexes" command.
+internal struct RenamedCollectionOperation<T: Codable>: Operation {
+    private var collection: MongoCollection<T>
+    private let to: String
+    private let dropTarget: Bool
+    private let options: RenamedCollectionOptions?
+
+    internal init(
+        collection: MongoCollection<T>,
+        to: String,
+        dropTarget: Bool,
+        options: RenamedCollectionOptions? = nil
+    ) {
+        self.collection = collection
+        self.to = to
+        self.dropTarget = dropTarget
+        self.options = options
+    }
+
+    internal func execute(using connection: Connection, session: ClientSession?) throws -> MongoCollection<T> {
+        let opts = try encodeOptions(options: options, session: session)
+        var error = bson_error_t()
+        return try self.collection.withMongocCollection(from: connection) { collPtr in
+            try withOptionalBSONPointer(to: opts) { optsPtr in
+                let success = mongoc_collection_rename_with_opts(
+                    collPtr,
+                    nil,
+                    self.to,
+                    self.dropTarget,
+                    optsPtr,
+                    &error
+                )
+
+                guard success else {
+                    throw extractMongoError(error: error)
+                }
+
+                let db = self.collection._client.db(self.collection.namespace.db)
+                let opts = self.collection.options
+                return MongoCollection(name: self.to, database: db, options: opts)
```

Comment:
our API is designed such that `MongoDatabase`s inherit various options (e.g. read concern) from their parent `MongoClient`, and so that `MongoCollection`s inherit those same options from their parent `MongoDatabase`. 

a user can override this by passing in a different setting to the `db` or `collection` method. right now I think we have a problem where database options might not be properly inherited. consider the following:
```swift
let db = client.db("db1", options: DatabaseOptions(readPreference: .secondary))
let coll = db.collection("coll1") // self.options will be nil, but `self.readPreference` will be `.secondary`, inherited rom db
let coll2 = coll.renamed(to: "coll2") // since we just copied what was in `self.options` and created a new `MongoDatabase` from scratch, we didn't copy the read preference
```

I think probably the easiest way to work around this is to introduce a new internal `copying` init to `MongoCollection` that takes in another collection and just copies over all of its properties but sets the namespace to the new namespace, e.g.

```swift
init(copying other: MongoCollection<T>, name: String) {
    self.namespace = MongoNamespace(db: other.namespace.db, collection: name)
    self.readPreference = other.readPreference
    // ...etc
}
```

and then you could use that here instead of creating the `db` and using it to call the old initializer. that way we know all the properties except the namespace will be exactly the same.

I think this would also eliminate the need to store the `options` property on `MongoCollection`

Reply:
alternatively you could update the existing initializer to update the passed-in options object to reflect inherited options, e.g. store `readPreference` on that so `coll1`'s options would reflect the actual read preference, but I'd lean toward option 1 since it cuts down on storing duplicate information